Cursor Testing

jerry-ng2 opened the pull request:
Cursor testing against SWE-bench dataset.

jerry-ng2 committed:
Testing Cursor against SWE-bench

Added scripts to help with setting up and capturing results of Cursor against SWE-bench dataset

jerry-ng2 committed:
Removed prompts.md and added requirements

jerry-ng2 committed:
Updated README.md

jerry-ng2 committed:
Updated README.md

jerry-ng2 committed:
Added swebench

jerry-ng2 committed:
Updated README.md and moved cursor_evaluation

suppathak commented on cursor-testing/README.md:
```diff
@@ -0,0 +1,222 @@
+
+## Overview
+
+SWE-bench-lite contains 300 real-world GitHub issues from popular Python repositories. This testing framework helps you:
```
Could you please add a hyperlink to the SWEB-bench-lite repository for easier access?

suppathak commented on cursor-testing/README.md:
````diff
@@ -0,0 +1,222 @@
+
+```bash
+# Install dependencies
+pip install -r requirements.txt
````
Suggest adding a step to create and activate a virtual Python environment before installing dependencies, this helps avoid conflicts with global packages.

suppathak commented on cursor-testing/README.md:
````diff
@@ -0,0 +1,222 @@
+
+1. **Open repository in Cursor**:
+   ```bash
+   cursor data/cursor_workspace/instance_001_repo_name/
````
```suggestion
   cursor data/cursor_workspace/<instance_repo_name>/
```

suppathak commented on cursor-testing/README.md:
````diff
@@ -0,0 +1,222 @@
+4. **Capture results**:
+   ```bash
+   # For successful completion
+   python capture_cursor_results.py --instance 1 --status completed --time 15 --difficulty medium --success success
````
Could you clarify how the values --instance 1 and --time 15 were determined?

- How did you arrive at 1—does it correspond to a specific issue or is it just an arbitrary choice?
- What does --time 15 represent—15 minutes, or something else?
- Where are these values defined or sourced from?

Also, how is "success" determined in this context, based on Cursor’s output matching ground truth, or a manual judgment?
[A screenshot of a successful run in the Cursor UI would be helpful for context.]

suppathak commented on cursor-testing/README.md:
````diff
@@ -0,0 +1,222 @@
+
+# Make sure you're in the SWE-bench directory
+cd /path/to/cursor-testing
+```
````
One thing I noticed, could you also clarify which model was selected within Cursor for code generation?
It would be helpful to mention whether you used GPT-4, Claude, Gemini, or a local model, and how the model was configured (e.g., any temperature or context settings if applicable).

Including this will make the evaluation more reproducible and help others understand the behavior you're observing.
[Screenshots would be helpful]

jerry-ng2 replied:
I added a section where I listed the models I used for this specific testing. I did not change any of the temperatures or context settings and thought that a screenshot might be too much.

suppathak reviewed (commented):
Thanks @jerry-ng2 . This is great start. 
I have added few comments. ptal.

Also, since this is still an early prototype for evaluating code gen, I would recommend generating just one instance file and including the evaluation report for that single case. This will help keep the PR concise and easier to review without generating too many result files up front. wdyt?

In your final report, you can simply mention that you evaluated n instances (which you can include in a follow-up PR) and summarize the broader trends or findings there.

jerry-ng2 committed:
Updated README.md

jerry-ng2 commented:
> Thanks @jerry-ng2 . This is great start. I have added few comments. ptal.
> 
> Also, since this is still an early prototype for evaluating code gen, I would recommend generating just one instance file and including the evaluation report for that single case. This will help keep the PR concise and easier to review without generating too many result files up front. wdyt?
> 
> In your final report, you can simply mention that you evaluated n instances (which you can include in a follow-up PR) and summarize the broader trends or findings there.

I plan to make a notebook with my entire analysis that is separate from the README.md. I am thinking of just keeping the cursor_predictions.jsonl that is just the summary of all the experiments and getting rid of the individual instances since those are created when the scripts are executed. Let me know what you think

jerry-ng2 committed:
Updated README.md

jerry-ng2 committed:
Removed instances and updated README.md

Removed all instances except 1 to condense PR and updated README.md

jerry-ng2 committed:
fixed failed image in README.md

jerry-ng2 committed:
Add comprehensive SWE-bench analysis plots and documentation assets

jerry-ng2 committed:
Add interactive Jupyter notebook for comprehensive cursor analysis

- Create cursor_analysis.ipynb with interactive data exploration capabilities
- Includes all visualizations: success rates, completion matrix, difficulty analysis
- Provides model comparison, repository analysis, and automated insights
- Generates exportable results in JSON and CSV formats
- Enables cell-by-cell analysis for deeper investigation

jerry-ng2 committed:
Updated cursor_analysis.ipynb

Refined the recommendations & Insights

jerry-ng2 committed:
Edited cursor_analysis.ipynb

Changed Key Insights section of notebook to give a deeper analysis of traits for each model

jerry-ng2 committed:
Changed Insights of notebook

Moved insights to a markdown instead of print statements

jerry-ng2 committed:
Re-ran Insight cell

jerry-ng2 committed:
Moved cursor_analysis.ipynb

Restructured root directory and moved cursor_analysis,ipynb to new directory docs/analysis

jerry-ng2 committed:
Added time chart

Added average time for models to resolve instances chart into notebook

jerry-ng2 committed:
Fixed time chart and linked notebook in README.md

suppathak commented on cursor-testing/README.md:
```diff
@@ -0,0 +1,275 @@
+# Optional: Setup 10 instances
+python cursor_swebench_tester.py --mode setup\
+   --start 0 \
+   --end 1
```
```suggestion
   --end 10
```
Since you are setting up 10 instances

suppathak commented on cursor-testing/README.md:
````diff
@@ -0,0 +1,275 @@
+   ```
+5. **Usage**:
+   ```bash
+   python capture_cursor_results --instance <instance_id> \
````
```suggestion
   python capture_cursor_results.py --instance <instance_id> \
```

suppathak commented on cursor-testing/README.md:
```diff
@@ -0,0 +1,275 @@
+- **Detailed logs**: Per-instance success/failure information
+- **Test results**: Whether fixes pass the original test suites
+
+Results are saved in `cursor.cursor_evaluation.json`
```
Can you add a hyperlink to this fine. I suppose it is in data/cursor_evaluation folder.

suppathak reviewed (commented):
Thank you @jerry-ng2 — great work! I have added a few minor nits for polish.

jerry-ng2 committed:
Updated README.md

suppathak commented:
/lgtm , thanks for addressing the comments @jerry-ng2 ! Merging this now.
